projects/netgrif-components-core/src/lib/authorization/role/role-guard.service.ts

```diff
@@ -37,22 +37,39 @@ export class RoleGuardService implements CanActivate {
     }
 
     public canAccessView(view: View, url: string): boolean {
-        if (typeof view.access !== 'string' && view.access.hasOwnProperty('role')) {
-            const allowedRoles = this.parseRoleConstraints(view.access.role, url);
-
-            return allowedRoles.some(constraint => {
-                if (constraint.roleIdentifier) {
-                    return this._userService.hasRoleByIdentifier(constraint.roleIdentifier, constraint.processIdentifier);
-                } else {
-                    return this._userService.hasRoleByName(constraint.roleName, constraint.processIdentifier);
-                }
-            });
+        if (typeof view.access !== 'string' && (view.access.hasOwnProperty('role') || view.access.hasOwnProperty('bannedRole'))) {
+
+            if (view.access.hasOwnProperty('bannedRole')) {
+                const bannedRoles = this.parseRoleConstraints(view.access.bannedRole, url);
+
+                return (bannedRoles.length == 0) || (bannedRoles.length > 0 && !bannedRoles.some(constraint => {
```

Comment:
if there are no banned roles, then the access rights should be determined by the `role` (if it exists). This implementation alwasy returns true with empty banned roles